test/Tests.Infrastructure/RavenExplicitDataAttribute.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using System.Reflection;
+using Xunit.Sdk;
+
+namespace Tests.Infrastructure;
+
+public class RavenExplicitDataAttribute : DataAttribute
+{
+    public RavenSearchEngineMode SearchEngineMode { get; set; }
+
+    public RavenDatabaseMode DatabaseMode { get; set; }
+
+    public object[] Data { get; set; } = null;
+
+    public RavenExplicitDataAttribute(
+        RavenDatabaseMode databaseMode = RavenDatabaseMode.Single,
+        RavenSearchEngineMode searchEngine = RavenSearchEngineMode.Lucene
+    )
+    {
+        DatabaseMode = databaseMode;
+        SearchEngineMode = searchEngine;
+    }
+
+    public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+    {
+        foreach (var (databaseMode, options) in RavenDataAttribute.GetOptions(DatabaseMode))
+        {
+            foreach (var (searchMode, o) in RavenDataAttribute.FillOptions(options, SearchEngineMode))
+            {
+                var length = 1;
+                if (Data is { Length: > 0 })
+                    length += Data.Length;
+
+                var array = new object[length];
+
+                array[0] = new RavenTestParameters
+                {
+                    SearchEngine = searchMode,
+                    DatabaseMode = databaseMode,
+                };
+
+                for (var i = 1; i < array.Length; i++)
+                    array[i] = Data[i - 1];
+
+                yield return array;
+            }
+        }
+    }
+}
+
+public class RavenTestParameters
```

Comment:
We have lost the options here.

Reply:
Yes. After talking to Pawel we think that we won't need options here since it's supposed to be used for low level tests.

Reply:
OK. Will rebase and adjust.